homeassistant/components/asuswrt/__init__.py

```diff
@@ -77,10 +78,16 @@
         dnsmasq=conf[CONF_DNSMASQ],
     )
 
-    await api.connection.async_connect()
-    if not api.is_connected:
-        _LOGGER.error("Unable to setup component")
-        return False
+    try:
+        await api.connection.async_connect()
+        if not api.is_connected:
+            _LOGGER.error("Unable to setup component")
```

Comment:
Instead of logging an error and return false, we could raise a PlatformNotReady here as well?

Reply:
This was original code. Maybe the case that really the connection on the router is disabled so platform must not load at all?

Reply:
It can be enabled again during the runtime of HA. It is an external factor that currently doesn't work, but might start working again.

Reply:
Ok, corrected as suggested